packages/language-picker/src/language-picker.tsx

```diff
@@ -77,27 +113,96 @@ const LanguagePicker = ( { onSelectLanguage, languages, languageGroups }: Props
-						<span className={ isSelected ? 'is-selected' : '' }>{ languageGroup.name() }</span>
-					</Button>
-				</div>
+				<Button
+					key={ languageGroup.id }
+					onClick={ onClick }
+					className="language-picker-component__language-group"
+				>
+					<span className={ isSelected ? 'is-selected' : '' }>{ languageGroup.name() }</span>
+				</Button>
 			);
 		} );
 	};
 
+	const shouldDisplayRegions = ! search;
+
+	const languagesToRender = search
+		? getSearchedLanguages( languages, search, localizedLanguageNames )
+		: getFilteredLanguages();
+
+	const selectControlOptions = languageGroups.map( ( lg ) => ( { key: lg.id, name: lg.name() } ) );
+
 	return (
-		<div className="language-picker-component">
-			<div className="language-picker-component__regions-label">
-				{ __( 'regions', __i18n_text_domain__ ) }
-			</div>
-			<div className="language-picker-component__content">
-				<div className="language-picker-component__category-filters">
-					{ renderCategoryButtons() }
+		<Flex className="language-picker-component" align="left">
+			<FlexBlock className="language-picker-component__heading">
+				<Flex justify="space-between" align="left">
+					<FlexItem className="language-picker-component__title wp-brand-font">
+						{ headingTitle || __( 'Select a language', __i18n_text_domain__ ) }
+					</FlexItem>
+					{ headingButtons && (
+						<FlexItem className="language-picker-component__heading-buttons">
+							{ headingButtons }
+						</FlexItem>
+					) }
+				</Flex>
+			</FlexBlock>
+			<FlexBlock className="language-picker-component__search">
+				<CustomSelectControl
+					label={ __( 'regions' ) }
+					hideLabelFromVision
+					value={ selectControlOptions.find( ( option ) => option.key === filter ) }
+					options={ selectControlOptions }
+					onChange={ ( { selectedItem } ) => selectedItem && setFilter( selectedItem.key ) }
+				/>
+				<div className="language-picker-component__search-mobile">
+					<Search
+						onSearch={ setSearch }
+						pinned
+						fitsContainer
+						placeholder={ __( 'Search languages…' ) }
+						onSearchClose={ () =>
+							setFilter(
+								findBestDefaultLanguageGroupId( selectedLanguage, languageGroups, filter )
+							)
+						}
+					/>
+				</div>
+				<div className="language-picker-component__search-desktop">
+					<Search
+						openIconSide="right"
+						onSearch={ setSearch }
+						fitsContainer
+						placeholder={ __( 'Search languages…' ) }
+						onSearchClose={ () =>
+							setFilter(
+								findBestDefaultLanguageGroupId( selectedLanguage, languageGroups, filter )
+							)
+						}
+					/>
 				</div>
+			</FlexBlock>
+			<FlexBlock className="language-picker-component__labels">
+				{ shouldDisplayRegions ? (
+					<>
+						<div className="language-picker-component__regions-label">{ __( 'regions' ) }</div>
+						<div className="language-picker-component__languages-label">{ __( 'languages' ) }</div>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **48** times:
`translate( 'Languages', { context: 'Help topic'} )` *ES Score: 10*